docs/proposals/mvp-dashboard-design.md

```diff
@@ -43,18 +43,125 @@ can query the relevant test builds.
+
+   Grafana lacks the ability to directly compare multiple metrics from
+   different test runs from different times. Since one of our goals is to create
+   a tool that can be used to compare test runs across different lab
+   environments it became clear that Grafana would not be able to fulfill our
+   demands.
+
+#### Alternatives
+
+- *Any proposed alternatives and their arguments should be located here.*
+
+### Choice of data storage
+
+1. **Need for SQL-like querying with optimizations**
+
+   [InfluxDB][1] is a database service optimized for temporal data, which is a
+   characteristic common to performance test data. The querying language
+   used is similar to SQL but incorporates time stamps and indexes off of
+   these whenever filtering. This type of optimization is a benefit to
+   the performance monitoring in this proposal because of the nature of
+   performance analysis.
+
+   Performance analysis requires the investigation of a series of metrics in a
+   test as they trend across time, this results in the high likelihood of
+   queried data being located close on the disk. These same time-based
+   reasoning applies for writing to disk, a write of performance metrics is
+   very likely followed by a similar metrics at a time interval ahead in time;
+   the implication being that the next write will occur close by on the disk.
+
+1. **Established database with historic performance data**
+
+   Previous performance data taken by the team is currently stored in an
+   InfluxDB server. Uniformity between our past, current, and future datasets
+   will ensure that we can make comparisons between different test results.
+   We can accomodate new and different data by properly using tags, a feature
+   of the InfluxDB query language.
 
 ## Implementation
 
-A description of the steps in the implementation, who will do them, and when.
+@meteorfox has volunteered to set up the scaffolding for both the server and
+the dashboard. The goal is to leave a foundation that can be handed over to
+@besanradwan and @vhe182 to work on developing the dashboard and server,
+respectively. Both of these team members will develop their system in tandem.
+These systems will use mocks for sending and receiving requests from the user,
+dashboard, server, or data storage.
+
+### Server Layout
+
+The server API will include those listed below and more will be added as the
+need for them arises during the development of the perf-dash. Server
+functionality requirements are to:
+
+- Make queries to data storage using the predefined queries in static files
+  and selected by the user.
+- Serve the data to the front-end.
+
+### Dashboard Layout
+
+bird
+
+## Open issues
+
+### Metric, Test, and Build Tagging
 
-## Open issues (if applicable)
+1. Currently, the performance data we have has not been tagged to distinguish
+   the different metrics, test type, or build run. To be able to query by
```

Comment:
It is with influx DB publisher for PKB. E.g. we have test type, timestamp, etc. However, the build run which presumably is linked to the release tag will be through jenkins. Maybe say this here as well.